examples/todo/main.go

```diff
@@ -0,0 +1,139 @@
+package main
```

Comment:
General comment: I think this is a really good example program, certainly the best we have for `script`. Actually, it may be a little _too_ good! The aim is to show how to use `script` to do useful things, but it's always difficult to write an example which shows what you want to show _and_ does something useful, without obscuring the main point.

You might like to take this program as is and give it its own repo (it's a useful tool!), and then slightly shorten this example to focus only on the `script` parts. For example, we could change this to just match the string `TODO` in any line, rather than detecting comment blocks, which is hard!

Reply:
I have simplified the example to only find todo existence in each line of code.